data_structures/linked_list/circular_doubly_linked_list.c

```diff
@@ -0,0 +1,230 @@
+{
+    struct node *n1;
+    n1 = (struct node *)malloc(sizeof(struct node));
+    n1->value = data;
+    if (*start == NULL)
+    {
+        n1->next = n1;
+        n1->prev = n1;
+        *start = n1;
+    }
+    else
+    {
+        struct node *temp1, *temp2;
+        temp1 = *start;
+        temp2 = (*start)->prev;
+        n1->prev = temp2;
+        n1->next = temp1;
+        temp1->prev = n1;
+        temp2->next = n1;
+    }
+    printf("\nNode succesfully inserted\n\n");
+}
+
+/**
+ * @brief   Deletion of the first node in list function
+ * @param   start   starting pointer of list
+ * @returns void
+ */
+void delete_first_node(struct node **start)
+{
+    if (*start == NULL)
+    {
+        printf("⚠️list is Empty⚠️\n");
+    }
+    else
+    {
+        struct node *temp1, *temp2;
+        temp1 = *start;
+        temp2 = temp1->prev;
+        if (temp1 == temp2)
+        {
+            free(temp2);
+            *start = NULL;
+        }
+        else
+        {
+            temp2->next = temp1->next;
+            (temp1->next)->prev = temp2;
+            *start = temp1->next;
+            free(temp1);
+            temp1 = NULL;
+        }
+        printf("\nNode succesfully deleted\n\n");
+    }
+}
+
+/**
+ * @brief   Deletion of the last node in list function
+ * @param   start   starting pointer of list
+ * @returns void
+ */
+void delete_last_node(struct node **start)
+{
+    if (*start == NULL)
+    {
+        printf("list is Empty\n");
+    }
+    else
+    {
+        struct node *temp1, *temp2;
+        temp1 = *start;
+        temp2 = temp1->prev;
+        if (temp1 == temp2)
+        {
+            free(temp2);
+            *start = NULL;
+        }
+        else
+        {
+            (temp2->prev)->next = temp1;
+            temp1->prev = temp2->prev;
+            free(temp2);
+            temp2 = NULL;
+        }
+        printf("\nNode succesfully deleted\n\n");
+    }
+}
+
+/**
+ * @brief   Display list function
+ * @param   start   starting pointer of list
+ * @returns void
+ */
+void Display(struct node *start)
+{
+    printf("\nIts Your Linked-list\n\n");
+    struct node *temp;
+    temp = start;
+    if (start != NULL)
+    {
+        while (temp->next != start)
+        {
+            printf("%lld -> ", temp->value);
+            temp = temp->next;
+        }
+        if (temp->next == start)
+        {
+            printf("%lld", temp->value);
+        }
+    }
+    else
+    {
+        printf("list is Empty\n");
+    }
+    printf("\n");
+}
+
+/**
+ * @brief Main function
+ * @returns 0 on exit
+ */
+int main()
+{
+    // Creating an empty list
+    struct node *start = NULL;
+    // Flg to stop loop and value to insert
+    int flag = 1, value;
+    while (flag == 1)
+    {
+        int ch;
+        printf("\nWhat do you want to perform?\n");
+        printf("ENTER YOUR CHOISE\n1. Insert At Start\n2. Insert At End\n3. DISPLAY LINKED LIST\n4. Delete First node\n5. Delete Last Node\n6. Exit\n");
```

Comment:
Shouldn't this be `choice` instead?
```suggestion
        printf("ENTER YOUR CHOICE\n1. Insert At Start\n2. Insert At End\n3. DISPLAY LINKED LIST\n4. Delete First node\n5. Delete Last Node\n6. Exit\n");
```

Reply:
Yeah 😅